internal/yggdrasil/yggdrasil.go

```diff
@@ -313,6 +314,12 @@ func (h *Handler) PostDataMessageForDevice(ctx context.Context, params yggdrasil
 		if err != nil {
 			return operations.NewPostDataMessageForDeviceInternalServerError()
 		}
+		if heartbeat.Hardware != nil {
+			err = h.deviceRepository.UpdateLabels(ctx, edgeDevice, hardware.MapLabels(heartbeat.Hardware))
```

Comment:
will this cause labels changes performed by the admin be overridden by next heartbeat?

Reply:
If there is the same key used it will override the value but other labels will be untouched.